### PR TITLE
Document use case of passing test dataloaders to Trainer.test()

### LINKED ISSUE #1990: Add Docs Demo'ing Test-Set Loaders With Trainer.Test()
## 📚 Documentation

I was wondering the best practice way to specify a new test_loader when using `trainer.test()`, similar to how we can for `trainer.fit()`. The code was already written, the docs just needed to be updated to sync with it.

### PR DESCRIPTION
**Pull request by authman**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? No, but it was discussed on the slack channel. I did make an issue for it as well #1990 
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes #1990 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Always 🙃

### CODE REVIEW

**Comment by authman** [comment]
Please note I based this off of `docs/changelog` as opposed to `master` because if I didn't, whoever merges it in would have had to deal with the merge conflicts. `docs/changelog` is currently ahead of master anyway, and this PR fits into that scope of work.

File: docs/source/test_set.rst [hunk 1]
```diff
@@ -34,4 +35,23 @@ To run the test set on a pre-trained model, use this method.
...
 
 In this  case, the options you pass to trainer will be used when
-running the test set (ie: 16-bit, dp, ddp, etc...)
\ No newline at end of file
+running the test set (ie: 16-bit, dp, ddp, etc...)
+
+
+Test additional data loaders
+----------------------------
+If no test sets were passed in to `trainer.fit()` or otherwise declared inside your
```
Comment (awaelchli) [inline docs/source/test_set.rst:43]: test sets can't be passed into trainer.fit
also, if they are declared in LightningModule, they get ignored and only the new dataloader(s) get used. I'm 99% sure.

File: docs/source/test_set.rst [hunk 2]
```diff
@@ -34,4 +35,24 @@ To run the test set on a pre-trained model, use this method.
...
+
+.. code-block:: python
+
+    # setup your data loader
+    test = DataLoader(...)
+
+    # test (pass in the loader)
+    trainer.test(test_dataloaders=test)
+
+You can either pass in either a single dataloader or a list of them. This
```
Comment (awaelchli) [inline docs/source/test_set.rst:56]:
```suggestion
You can pass in either a single dataloader or a list of them. This
```

File: docs/source/test_set.rst (hunk 1 above)
Comment (Borda) [reply docs/source/test_set.rst:43]: https://github.com/PyTorchLightning/pytorch-lightning/blob/8ee6d91d0e8ea83829a04d35e7fe7500eb332211/pytorch_lightning/trainer/trainer.py#L779-L784

Comment (Borda) [reply docs/source/test_set.rst:43]: @awaelchli is right here...

Comment (awaelchli) [reply docs/source/test_set.rst:43]: where? I only see val dataloaders

Comment (authman) [reply docs/source/test_set.rst:43]: This is my mistake. In my brain, I still had the code base for old version of PTL when test_dataloaders was indeed an argument passed into `.fit`: [see here](https://github.com/PyTorchLightning/pytorch-lightning/blob/ee68d5ba8e66bba52b3cc3d9d8780d6df2f3204b/pytorch_lightning/trainer/trainer.py#L602). But toady that is no longer the case.

Comment (Borda) [reply docs/source/test_set.rst:43]: I say that @awaelchli is correct, test set cannot be passed to `.fit()`

**Comment by Borda** [comment]
@authman so we can close this PR... :raccoon:

**Comment by awaelchli** [comment]
I think it would still be good to have the information to pass in dataloaders to Trainer.test

File: docs/source/test_set.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source/test_set.rst:43]: A slight reformulation would do it I think.

**Comment by Borda** [comment]
@authman mind reformate it for `test` as @awaelchli suggested?

**Comment by authman** [comment]
Working on the commit.

File: docs/source/test_set.rst [hunk 3]
```diff
@@ -34,4 +35,22 @@ To run the test set on a pre-trained model, use this method.
...
 
 In this  case, the options you pass to trainer will be used when
-running the test set (ie: 16-bit, dp, ddp, etc...)
\ No newline at end of file
+running the test set (ie: 16-bit, dp, ddp, etc...)
+
+
+Test additional data loaders
+----------------------------
+You can still run inference on a test set even if the `test_dataloader` method hasn't been
```
Comment (Borda) [inline docs/source/test_set.rst:43]:
```suggestion
You can still run inference on a test set even if the `test_dataloaders` method hasn't been
```

File: docs/source/test_set.rst [hunk 4]
```diff
@@ -34,4 +35,22 @@ To run the test set on a pre-trained model, use this method.
...
+defined within your LightningModule instance. This would be the case when your test data
```
Comment (Borda) [inline docs/source/test_set.rst:44]:
```suggestion
defined within your :class:`~pytorch_lightning.core.LightningModule` instance. This would be the case when your test data
```

File: docs/source/test_set.rst [hunk 5]
```diff
@@ -34,4 +35,22 @@ To run the test set on a pre-trained model, use this method.
...
 
 In this  case, the options you pass to trainer will be used when
-running the test set (ie: 16-bit, dp, ddp, etc...)
\ No newline at end of file
+running the test set (ie: 16-bit, dp, ddp, etc...)
+
+
+Test additional data loaders
+----------------------------
+You can still run inference on a test set even if the `test_dataloaders` method hasn't been
```
Comment (awaelchli) [inline docs/source/test_set.rst:43]:
```suggestion
You can still run inference on a test set even if the `test_dataloader` method hasn't been
```

File: docs/source/test_set.rst [hunk 6]
```diff
@@ -34,4 +35,22 @@ To run the test set on a pre-trained model, use this method.
     trainer.test(model)
 
 In this  case, the options you pass to trainer will be used when
-running the test set (ie: 16-bit, dp, ddp, etc...)
\ No newline at end of file
+running the test set (ie: 16-bit, dp, ddp, etc...)
+
+
+Test additional data loaders
+----------------------------
```
Comment (awaelchli) [inline docs/source/test_set.rst:42]:
```suggestion
Test with additional data loaders
---------------------------------
```
(since we are not testing the dataloaders themselves, but the data they provide)

File: docs/source/test_set.rst (hunk 3 above)
Comment (awaelchli) [reply docs/source/test_set.rst:43]: I changed it back because the LightningModule methods is singular (even if you can return multiple)

Comment (awaelchli) [reply docs/source/test_set.rst:43]: hope it's ok

File: docs/source/test_set.rst (hunk 4 above)
Comment (awaelchli) [reply docs/source/test_set.rst]: nice)
